apps/ui/BaseWindow.cpp

```diff
@@ -221,29 +225,54 @@ void BaseWindow::display( )
     _brayns->getCamera().commit();
     _brayns->render( renderInput, renderOutput );
 
-    GLenum format = GL_RGBA;
-    GLenum type   = GL_FLOAT;
-    GLvoid* buffer = 0;
-    switch(_frameBufferMode)
+    if( _brayns->getParametersManager().getApplicationParameters().getFilters().empty( ))
     {
-    case BaseWindow::FRAMEBUFFER_COLOR:
-        type = GL_UNSIGNED_BYTE;
-        buffer = renderOutput.colorBuffer.data( );
-        break;
-    case BaseWindow::FRAMEBUFFER_DEPTH:
-        format = GL_LUMINANCE;
-        buffer = renderOutput.depthBuffer.data( );
-        break;
-    default:
-        glClearColor(0.f,0.f,0.f,1.f);
-        glClear( GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT );
+
+        GLenum format = GL_RGBA;
+        GLenum type   = GL_FLOAT;
+        GLvoid* buffer = 0;
+        switch(_frameBufferMode)
+        {
+        case BaseWindow::FRAMEBUFFER_COLOR:
+            type = GL_UNSIGNED_BYTE;
+            buffer = renderOutput.colorBuffer.data( );
+            break;
+        case BaseWindow::FRAMEBUFFER_DEPTH:
+            format = GL_LUMINANCE;
+            buffer = renderOutput.depthBuffer.data( );
+            break;
+        default:
+            glClearColor(0.f,0.f,0.f,1.f);
+            glClear( GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT );
+        }
+
+        _fps.stop();
+        glDrawPixels( _windowSize.x( ), _windowSize.y( ), format, type, buffer );
     }
+    else
+    {
+        ScreenSpaceProcessorData ssProcData;
+
+        ssProcData.width = _windowSize.x( );
+        ssProcData.height = _windowSize.y( );
+
+        ssProcData.colorFormat = GL_RGBA;
+        ssProcData.colorBuffer = renderOutput.colorBuffer.data( );
+        ssProcData.colorType = GL_UNSIGNED_BYTE;
 
-    _fps.stop();
+        ssProcData.depthFormat = GL_LUMINANCE;
+        ssProcData.depthBuffer = renderOutput.depthBuffer.data( );
+        ssProcData.depthType = GL_FLOAT;
+
+        _screeSpaceProcessor.draw( ssProcData );
+
+        _fps.stop();
```

Comment:
I think the _fps.stop() can be moved down out of the scope, and generalized for both cases.